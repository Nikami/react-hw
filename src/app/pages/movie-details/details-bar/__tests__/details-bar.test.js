import React from 'react';
import { shallow } from 'enzyme';
import DetailsBar from '../index';

describe('DetailsBar', () => {
  it('matches snapshot', () => {
    const defaultProps = {
      title: 'Test',
      vote_count: 1,
      genres: ['Genre1', 'Genre2'],
      release_date: '2020-05-01',
      poster_path: 'https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg',
      overview: 'Test',
    };
    const wrapper = shallow(<DetailsBar movie={defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
