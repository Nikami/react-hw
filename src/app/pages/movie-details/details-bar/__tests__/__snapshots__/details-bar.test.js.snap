// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsBar matches snapshot 1`] = `
<WithStyles(BgContainer)>
  <div
    className="details-header"
  >
    <_default />
    <Link
      className="details-header__link"
      to="/"
    >
      <WithStyles(Button)
        color="primary"
        size="small"
        variant="contained"
      >
        Search
      </WithStyles(Button)>
    </Link>
  </div>
  <div
    className="details-bar"
  >
    <img
      alt="Test"
      className="details-bar__poster"
      src="https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg"
    />
    <div
      className="details-bar__info"
    >
      <div
        className="details-bar__rate"
      >
        <WithStyles(Typography)
          className="details-bar__title"
          component="h3"
          variant="h6"
        >
          Test
        </WithStyles(Typography)>
        <div
          className="details-bar__rating"
        >
          1
        </div>
      </div>
      <WithStyles(Typography)
        className="details-bar__genres"
        component="h6"
        variant="subtitle2"
      >
        Genre1 & Genre2
      </WithStyles(Typography)>
      <WithStyles(Typography)
        className="font-bold"
        component="h6"
        gutterBottom={true}
        variant="subtitle2"
      >
        2020
      </WithStyles(Typography)>
      <WithStyles(Typography)
        className="details-bar__overview"
        component="p"
        gutterBottom={true}
        variant="subtitle1"
      >
        Test
      </WithStyles(Typography)>
    </div>
  </div>
</WithStyles(BgContainer)>
`;
